### ISSUE TITLE
Add the Iterator Symbol to functions which return nodes (ex. cts.search)

### ISSUE THREAD

**Issue by sjordan1975** [grtjn/marklogic-typescript-definitions#16]
Here is a snippet of JS that works in QConsole:

```JavaScript
let results = [];

var items = cts.search(
    cts.andQuery([
      cts.directoryQuery("/searchable/"),
      cts.elementWordQuery(
      xs.QName("maintenance-committee"),
        "AV-003", "case-insensitive")]));

for (item of items){
   var value = item.xpath("/searchable/db-meta/maintenance-committee/string()");

   results.push(value);
}
```

Here is the same code written in TS:

```TypeScript
let results = [];

let items = cts.search(
    cts.andQuery([
      cts.directoryQuery("/searchable/"),
      cts.elementWordQuery(
      xs.QName("maintenance-committee"),
        “AV-003", "case-insensitive")])).toArray();

for (let item of items){
    let value: string = item.xpath("/searchable/db-meta/maintenance-committee/string()”);

results.push(value);

}
```

Without the `.toArray()`, the resulting JS output from the TS transpiler doesn’t work.

**Comment by grtjn** [comment]
@christyharagan, what do you make of this? Is the transpiler just doing a bad job, or is there something more serious going on?

**Comment by jmakeig** [comment]
You should never use `toArray()` when you don’t have to. The whole point of the iterator (i.e. what you get by calling the `Symbol.iterator` method) is to not have to eagerly evaluate all of the results. `cts.search()` returns a `ValueIterator` (in MarkLogic 8) which is (confusingly) bother an iterator and and iterable. I don't know the TypeScript-specific issue here, but [here’s a polyfill for ES2016 `Map` type](https://github.com/Microsoft/TypeScript/issues/3069#issuecomment-99964139) that exposes its interface.

**Comment by christyharagan** [comment]
No, it was just poor typing effort on my behalf. It was a good catch by Salim. I'll put a fix in...

**Comment by grtjn** [comment]
Christy probably fixed this in one of the 0.3.0 to 0.4.2 releases, but it has now moved over to Sequence. Hopefully that works equally well..

**Comment by grtjn** [comment]
@sjordan1975 Would you be able to run a test with the current code in this repo? It is tagged with 0.5.0, but not yet released on NPM..
